workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestQuery.cs

```diff
@@ -0,0 +1,117 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct ComponentInterest.Query objects.
+    /// </summary>
+    public class InterestQuery
+    {
+        private ComponentInterest.Query query;
+
+        // Hides the default constructor
+        private InterestQuery()
+        {
+        }
+
+        /// <summary>
+        ///     Creates an InterestQuery.
+        /// </summary>
+        /// <param name="constraint">
+        ///     A Constraint object defining the constraints of the query.
```

Comment:
You could make `Constraint` a `<cref>` to allow cross-linking to that type's reference docs (same for the return type)